intake_esm/source.py

```diff
@@ -43,18 +43,25 @@ def _open_dataset(
 ):
 
     _can_be_local = fsspec.utils.can_be_local(urlpath)
-    storage_options = xarray_open_kwargs['backend_kwargs'].get('storage_options', {})
+    storage_options = xarray_open_kwargs.get('backend_kwargs', {}).get('storage_options', {})
     if xarray_open_kwargs['engine'] == 'zarr':
         url = urlpath
     elif _can_be_local:
         url = fsspec.open_local(urlpath, **storage_options)
     else:
         url = fsspec.open(urlpath, **storage_options).open()
 
-    ds = xr.open_dataset(url, **xarray_open_kwargs)
+    # Handle multi-file datasets with `xr.open_mfdataset()`
+    if '*' in url or isinstance(url, list):
+        # How should we handle concat_dim, and other xr.open_mfdataset kwargs?
+        xarray_open_kwargs.update(preprocess=preprocess)
+        xarray_open_kwargs.update(parallel=True)
```

Comment:
@agstephens, do you have working end-to-end examples that are using this glob pattern? Are you passing `concat_dim` and other `xr.open_mfdataset` via the `xarray_open_kwargs` in parameter `cat.to_dataset_dict()`?

Reply:
> strangely I have noticed that github did not seem to be up-to-date with what I have on the file system. I got these tests working:

I  previously broke the main branch of your fork. I was able to recover your changes by creating a [new branch](https://github.com/intake/intake-esm/tree/agstephens/main) from the main repository. 

> @agstephens, do you have working end-to-end examples that are using this glob pattern? Are you passing concat_dim and other xr.open_mfdataset via the xarray_open_kwargs parameter in cat.to_dataset_dict() call?

I'm going to test this.  There's a chance we need to take a different code path during the concatenation stage when `open_mfdataset` is used.

Reply:
It appears that things seem to be working fine without needing additional changes... I'm going to merge this once the CI passes. In case something else is broken, we shall revisit this in a separate PR. 


Thank you for putting this together, @agstephens!